modules/index/src/main/java/org/apache/ignite/internal/index/IndexManagementUtils.java

```diff
@@ -0,0 +1,266 @@
+     * performing the operation and the inability to complete it due to execution in the metastore thread or on recovery (the metastore
+     * watches will not be deployed yet). Logs errors if it is not {@link IndexNotFoundValidationException},
+     * {@link IndexAlreadyAvailableValidationException} or {@link NodeStoppingException}.
+     *
+     * @param catalogManager Catalog manger.
+     * @param indexId Index ID.
+     * @param log Logger.
+     */
+    static void makeIndexAvailableInCatalogWithoutFuture(CatalogManager catalogManager, int indexId, IgniteLogger log) {
+        catalogManager
+                .execute(MakeIndexAvailableCommand.builder().indexId(indexId).build())
+                .whenComplete((unused, throwable) -> {
+                    if (throwable != null) {
+                        Throwable unwrapCause = unwrapCause(throwable);
+
+                        if (!(unwrapCause instanceof IndexNotFoundValidationException)
+                                && !(unwrapCause instanceof IndexAlreadyAvailableValidationException)
+                                && !(unwrapCause instanceof NodeStoppingException)) {
+                            log.error("Error processing the command to make the index available: {}", unwrapCause, indexId);
+                        }
+                    }
+                });
+    }
+
+    /**
+     * Extracts a partition ID from the key: {@value PARTITION_BUILD_INDEX_KEY_PREFIX} + {@code "<indexId>.<partitionId>"}.
+     *
+     * @param key Key.
+     * @return Partition ID.
+     */
+    static int extractPartitionIdFromPartitionBuildIndexKey(String key) {
+        assert key.startsWith(PARTITION_BUILD_INDEX_KEY_PREFIX) : key;
+
+        String[] strings = key.split("\\.");
+
+        return Integer.parseInt(strings[3]);
+    }
+
+    /**
+     * Extracts a index ID from the key: {@value PARTITION_BUILD_INDEX_KEY_PREFIX} + {@code "<indexId>.<partitionId>"}.
+     *
+     * @param key Key.
+     * @return Index ID.
+     */
+    static int extractIndexIdFromPartitionBuildIndexKey(String key) {
+        assert key.startsWith(PARTITION_BUILD_INDEX_KEY_PREFIX) : key;
+
+        String[] strings = key.split("\\.");
+
+        return Integer.parseInt(strings[2]);
+    }
+
+    /**
+     * Returns {@code true} if the local node is no longer the primary replica at the timestamp of interest.
+     *
+     * @param primaryReplicaMeta Primary replica meta.
+     * @param localNode Local node.
+     * @param timestamp Timestamp of interest.
+     */
+    static boolean isLeaseExpire(ReplicaMeta primaryReplicaMeta, ClusterNode localNode, HybridTimestamp timestamp) {
+        // TODO: IGNITE-20678 We need to compare by IDs: localNode.id().equals(primaryReplicaMeta.getLeaseholderId())
+        return !localNode.name().equals(primaryReplicaMeta.getLeaseholder()) && timestamp.after(primaryReplicaMeta.getExpirationTime());
```

Comment:
Let's ask @sanpwc

Reply:
In given case both options will work properly and thus I'd rather use compareTo because it allows to proceed with index build process a little bit longer.

Generally, it depends on what you need.
In case of using after() within comparison you will have cluster wide decision of whether lease is expired or not. Meaning that if any node X is saying that lease is expired then any other node will also say that lease is expired, despite the fact that for some specific nodes it'll be possible to have local time less then lease expiration timestamp.

In case of  using simple compareTo it'll be local decision of whether lease is expired or not.